### PR TITLE
feature: implement alert bulking for HTTP exporter

### PR DESCRIPTION
**Pull request by slashben**
This pull request introduces the alert bulking feature for the node-agent, enabling runtime alerts from the same container to be batched together before being sent to the Kubescape synchronizer. This reduces HTTP overhead, improves performance, and maintains backward compatibility. The implementation includes a new bulk manager, configuration options, process tree merging, container lifecycle integration, and comprehensive unit tests.

**Alert Bulking Implementation**

* New bulk manager: Added `AlertBulkManager` in `pkg/exporters/alert_bulk_manager.go` to manage per-container alert bulks, with configurable size and timeout-based flush triggers, and immediate flush on container stop.
* Process tree merging: Integrated process tree merging logic to combine trees from multiple alerts using utilities in `pkg/utils/processtree_merge.go`.

**Configuration and Integration**

* Bulking configuration: Extended `HTTPExporterConfig` to support enabling/disabling bulking, bulk size, and timeout parameters.
* Container lifecycle: Integrated bulk flushing with container removal events and agent shutdown to ensure all pending alerts are sent.

**Testing**

* Unit tests: Added `pkg/exporters/alert_bulk_manager_test.go` with thorough tests for bulk creation, flushing logic, container isolation, and edge cases.

**Documentation**

* New documentation: Created `docs/ALERT_BULKING.md` detailing architecture, requirements, configuration, and usage of the alert bulking feature.

### CODE REVIEW

File: pkg/exporters/alert_bulk_manager.go [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+	defer cb.Unlock()
+
+	alerts := cb.alerts
+	// ProcessTree struct wraps Process in .ProcessTree field
+	processTree := apitypes.ProcessTree{
+		ProcessTree: cb.mergedProcessTree,
+	}
+	cloudServices := cb.cloudServices
+
+	// Reset for potential reuse (though typically bulk is removed after flush)
```
Comment (matthyx) [inline pkg/exporters/alert_bulk_manager.go]: I wonder if that's really doing anything?

File: pkg/exporters/http_exporter.go [hunk 2]
```diff
@@ -128,11 +149,39 @@ func (config *HTTPExporterConfig) Validate() error {
 		config.QueryParams = []HTTPKeyValues{}
 	}
 
+	// Set defaults for bulking configuration
```
Comment (matthyx) [inline pkg/exporters/http_exporter.go]: this can be moved directly to the config package

File: pkg/utils/processtree_merge.go [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+		parent.ChildrenMap[apitypes.CommPID{PID: inserted.PID}] = inserted
+	} else {
+		// Parent doesn't exist yet - this will be linked when parent is inserted
+		// For now, if this is a root-level process in source tree, we need to decide how to handle it
+		// In practice, the parent should eventually be merged as we traverse the source tree
+	}
+}
+
+// copyUint32Ptr creates a copy of a uint32 pointer
+func copyUint32Ptr(src *uint32) *uint32 {
```
Comment (matthyx) [inline pkg/utils/processtree_merge.go]: hmm, we need that?

**Review by matthyx — APPROVED** [review]
